src/io/write_all.rs

```diff
@@ -0,0 +1,85 @@
+use core::mem;
+
+use io::Write;
+
+/// Creates a helper that will write all data from the provided buffer into the provided writer.
+///
+/// Once the buffer has been written the helper will return the writer + buffer,
+///
+/// In the case of an error from the writer the helper will discard both writer and buffer and
+/// return the error.
+///
+/// ```
+/// # #[macro_use]
+/// # extern crate nb;
+/// # fn main () {
+/// use nb::io::write_all;
+///
+/// let mut buffer = [0; 3];
+/// {
+///     let writer = &mut &mut buffer[..];
```

Comment:
`&mut &mut`?

Reply:
Remove one, it does _become_ an `&mut &mut` during the method calls, but it doesn't need to be one to pass into `write_all`.